trello-library/src/main/java/pl/flyingoctopus/trello/service/TrelloMockService.java

```diff
@@ -1,15 +1,29 @@
 package pl.flyingoctopus.trello.service;
 
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
 import reactor.core.publisher.Mono;
 
 @Component
 @RequiredArgsConstructor
 public class TrelloMockService implements TrelloService {
+
+    private final WebClient webClient = WebClient.builder().baseUrl("https://api.trello.com/1").build();
+    private final String key = "TEMPORARY_CONSTANT";
```

Comment:
To myślę że można przenieść do properties, tam mamy już `trello.token` to można dodać `trello.key`

Reply:
Wrzuciłem do properties `trello.key`, w TrelloMockService dodałem te trzy linijki
`private final TrelloProperties trelloProperties;`
`private final String key = trelloProperties.getKey();`
`private final String token = trelloProperties.getToken();`

Ale teraz jak próbuję zbudować cały projekt (`gradlew clean build`) to dostaję error 

> error: variable trelloProperties might not have been initialized

Wydaje mi się że problem jest z tym nagłówkiem `@ConfigurationProperties("trello")` w TrelloProperties, ale nie jestem do końca pewien co zrobić z tym errorem

Reply:
Te dwie linijki w zasadzie opisują wszystko:
```
/builds/dyedmagenta/octopus-services/trello-library/src/main/java/pl/flyingoctopus/trello/service/TrelloMockService.java:18: error: variable trelloProperties might not have been initialized
    private final String key = trelloProperties.getKey();
```
Sypie się dlatego, że w zmiennej `final` używasz zmiennej, która może nie być zainicializowana.


Weźmy taki przykład
```java
public class Example {
    private final String foo;
    private final String bar = foo + "bar";
    private static final String SOME_OTHER_FOO = "Foo";

    public Example(String foo) {
        this.foo = foo;
    }
}
```
W takim przypadku kompilator wywali ci błąd, ponieważ inicializujesz bar, zanim foo zostanie zainicializowane.
Modifier `final` wymusza na tobie inicializację zmiennej, zanim skończy wykonywać się konstruktor. 
W przykładzie kompilator wyłapuje, że zmienna foo jest używana do inicializacji bar. Jeśli chodzi o final to inicializacje w konstruktorze wykonują się jako ostatnie.

Czyli kolejność jest taka:
1. Pola statyczne - `private static final String SOME_OTHER_FOO`
2. Inicializacja pola - `private final String bar = foo + "bar";`
3. Konstruktor - `private final String foo;`

Więc jak masz taką sytuację:
```java
@Component
@RequiredArgsConstructor
public class TrelloMockService implements TrelloService {
    private final TrelloProperties trelloProperties;
    private final String key = trelloProperties.getKey();
}
```
To próbujesz inicializować `String key` zanim zostanie wywołany konstruktor, w którym jest inicializowany ` TrelloProperties trelloProperties`.
Dla przypomnienia `@RequiredArgsConstructor` - tworzy konstruktor dla wszystkich wymaganych pól, czyli `final` których jeszcze do tej pory nie zainicializujesz.
Jako że jest to Spring Bean `@Component` to spring podczas uruchomienia utworzy tę klasę i poda do konstruktora wymagane zależności.